WebApplicationDemo/Program.cs

```diff
@@ -33,8 +34,8 @@ public static IHostBuilder Config(this IHostBuilder builder, string[] args)
 			return builder
 				.ConfigureServices(s =>
 				{
-					s.AddSingleton<IExternalInjected, ExternalInjected>();
-				})
+					s.AddSingleton<IInitializeDbInjected>(new InitializeDbInjected());
```

Comment:
new here is wrong for sure. Not clear what is this for.

Reply:
even worse, this is _totally_ wrong.
this is not how you initialize the DB on application start but placing initialization code in the ctor that is called to Add a Singleton empty interface with no purpose.

the ctor is executed before adding the instance as singleton and never used. is pure garbage.

also, this removes the previously ExternalInjected example on how to inject services from Microsoft DI into the Container which was the purpose of IExternalInjected example.

Please restore the original code and move DB init to the proper place.

Reply:
I'm unsure what the correct methodology is.

Reply:
the correct methodology in general is not to and do this on the local testing project as you don't want to run this on a deployed service every time.

in the special case of this Sample, placing this in an IHostedService Start() would be best. 
Alternatively in the Main() before starting the Host would be acceptable too.